tools/src/main/scala/caliban/tools/ClientWriter.scala

```diff
@@ -124,10 +125,9 @@ object ClientWriter {
       .map(name => name.toLowerCase -> name)
       .groupBy(_._1)
       .collect {
-        case (_, (_ :: typeNamesToRename)) if typeNamesToRename.nonEmpty =>
-          typeNamesToRename.zipWithIndex.map { case (((_, originalTypeName), index)) =>
-            val suffix = "_" * (index + 1)
-            originalTypeName -> s"$originalTypeName$suffix"
+        case (_, _ :: typeNamesToRename) if typeNamesToRename.nonEmpty =>
+          typeNamesToRename.map { case (_, originalTypeName) =>
+            originalTypeName -> s"`$originalTypeName`"
```

Comment:
Ah, the problem was here actually. The old logic was adding `_1`, `_2`, etc to discriminate names but the new logic isn't anymore. I will revert that part.